Cabal/doc/developing-packages.rst

```diff
@@ -1,83 +1,119 @@
 
--  ``Setup.hs``: usually containing a few standardized lines of code,
-   but can be customized if necessary.
 
-We can create both files manually or we can use ``cabal init`` to create
-them for us.
+Using ``cabal init``
+--------------------
 
-Using "cabal init"
-------------------
+The ``cabal init`` command creates the necessary files for a Cabal package,
+it has both an ``--interactive`` (default) and ``--non-interactive``
+mode. The interactive mode will walk you through many of the package
+options and metadata, the non-interactive mode will simply pick reasonable
+defaults which is sufficient if you're just spinning up a small project to
+try something out.
 
-The ``cabal init`` command is interactive. It asks us a number of
-questions starting with the package name and version.
+.. highlight:: console
 
 ::
 
-    $ cabal init
-    Package name [default "proglet"]?
-    Package version [default "0.1"]?
-    ...
+    $ cabal init --non-interactive
+    # You can also use -n which is the short version of --non-interactive
 
-It also asks questions about various other bits of package metadata. For
-a package that you never intend to distribute to others, these fields
-can be left blank.
+If you want, you can also try out the interactive mode, for now chose
+"Executable" when asked what type of package you want to build.
 
-One of the important questions is whether the package contains a library
-or an executable. Libraries are collections of Haskell modules that can
-be re-used by other Haskell libraries and programs, while executables
-are standalone programs.
+.. highlight:: console
 
 ::
 
+    $ cabal init --interactive
+    ...
     What does the package build:
-       1) Library
-       2) Executable
+      1) Executable
+      2) Library
+      3) Library and Executable
     Your choice?
 
-For the moment these are the only choices. For more complex packages
-(e.g. a library and multiple executables or test suites) the ``.cabal``
-file can be edited afterwards.
 
-Finally, ``cabal init`` creates the initial ``proglet.cabal`` and
-``Setup.hs`` files, and depending on your choice of license, a
-``LICENSE`` file as well.
+This will create the ``Main.hs`` Haskell file and a few others that are
+part of all Cabal packages. By default ``cabal init`` will create an
+executable in the top level project directory.
 
 ::
 
-    Generating LICENSE...
     Generating Setup.hs...
+    Generating CHANGELOG.md...
+    Generating Main.hs...
     Generating proglet.cabal...
 
-    You may want to edit the .cabal file and add a Description field.
+Check out the files created using the ``ls`` command.
+
+::
+
+   $ ls
+   CHANGELOG.md  Main.hs  proglet.cabal  Setup.hs
+
+
+Running the program
+-------------------
+
+Now that we have our Haskell code and the extra files that Cabal needs we
+can build and run our project.
+
+::
+
+   $ cabal build
+   Resolving dependencies...
+   ...
+   Linking /path/to/proglet ...
+
+   $ cabal run
```

Comment:
I'd add the executable name. It's nice that `cabal run` works without it, but that is rare special case.

Reply:
Done.